### PR TITLE
[fabric] update to vault v2

### PR DESCRIPTION
**Pull request by alvaropicazo**
**Changelog**
- Updated commands for vault v2 version in every role where it was used.
- Updated charts with the correct paths required for v2 version. 

 - Merged develop code into vault v2 branch.

**Reviewed by**
@jagpreetsinghsasan 
@sownak 

 

**Linked issue**
#1317

### CODE REVIEW

**Comment by jagpreetsinghsasan** [comment]
@alvaropicazo squash merge the commits and sign the commit, the DCO is failing

File: platforms/hyperledger-fabric/charts/invoke_chaincode/templates/invoke_chaincode.yaml [hunk 1]
```diff
@@ -144,7 +144,7 @@ spec:
             echo $ENDORSING_ORG_NAMES
             for ns in $ENDORSING_ORG_NAMES
             do
-              SECRETPREFIX=secret/crypto/peerOrganizations/${CREATOR_NS}/endorsers/${ns}/users/admin
+              SECRETPREFIX=secretsv2/data/crypto/peerOrganizations/${CREATOR_NS}/endorsers/${ns}/users/admin
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/charts/invoke_chaincode/templates/invoke_chaincode.yaml:147]: I think fetching the `secret or secretv2` prefix from network.yaml is better.
Lets have a discussion around this today.

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 2]
```diff
@@ -191,7 +191,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:194]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 3]
```diff
@@ -278,7 +278,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:281]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 4]
```diff
@@ -366,7 +366,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:369]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 5]
```diff
@@ -451,7 +451,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:454]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml [hunk 6]
```diff
@@ -537,7 +537,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-new-channel.yaml:540]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 7]
```diff
@@ -154,7 +154,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:157]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 8]
```diff
@@ -240,7 +240,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:243]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 9]
```diff
@@ -315,7 +315,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:318]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 10]
```diff
@@ -403,7 +403,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:406]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 11]
```diff
@@ -488,7 +488,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:491]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml [hunk 12]
```diff
@@ -574,7 +574,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-ordererorg.yaml:577]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 13]
```diff
@@ -145,7 +145,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:148]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 14]
```diff
@@ -225,7 +225,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:228]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 15]
```diff
@@ -313,7 +313,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:316]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 16]
```diff
@@ -398,7 +398,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:401]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml [hunk 17]
```diff
@@ -484,7 +484,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabric-add-organization.yaml:487]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv-add-peer.yaml [hunk 18]
```diff
@@ -113,7 +113,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv-add-peer.yaml:116]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv-add-peer.yaml [hunk 19]
```diff
@@ -228,7 +228,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv-add-peer.yaml:231]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft-add-orderer.yaml [hunk 20]
```diff
@@ -149,7 +149,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft-add-orderer.yaml:152]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 21]
```diff
@@ -154,7 +154,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:157]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 22]
```diff
@@ -241,7 +241,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:244]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 23]
```diff
@@ -329,7 +329,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:332]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 24]
```diff
@@ -414,7 +414,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:417]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml [hunk 25]
```diff
@@ -500,7 +500,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2-raft.yaml:503]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml [hunk 26]
```diff
@@ -136,7 +136,7 @@ network:
       vault:
         url: "vault_addr"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml:139]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml [hunk 27]
```diff
@@ -216,7 +216,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml:219]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml [hunk 28]
```diff
@@ -303,7 +303,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml:306]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml [hunk 29]
```diff
@@ -387,7 +387,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml:390]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml [hunk 30]
```diff
@@ -472,7 +472,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml:475]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml [hunk 31]
```diff
@@ -104,7 +104,7 @@ network:
       vault:
         url: "vault_url"
         root_token: "vault_root_token"
-        secret_path: "secret"
+        secret_path: "secretsv"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml:107]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml [hunk 32]
```diff
@@ -174,7 +174,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml:177]: `secretsv` or `secretsv2` ?

File: platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml [hunk 33]
```diff
@@ -257,7 +257,7 @@ network:
...
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/samples/network-proxy-none.yaml:260]: `secretsv` or `secretsv2` ?

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-fabric/configuration/roles/create/crypto/orderer/tasks/orderer.yaml [hunk 34]
```diff
@@ -10,7 +10,7 @@
   changed_when: false
   tags:
     - no-idempotence-test
-
+
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/create/crypto/orderer/tasks/orderer.yaml:13]: The following task should be changed as not vault_msp_result will have only one response. Please add a new task to get the complete msp secret and then pass that variable in vault_output for setup/get_crypto role

File: platforms/hyperledger-fabric/configuration/roles/create/crypto/orderer/tasks/orderer.yaml [hunk 35]
```diff
@@ -172,7 +172,7 @@
  #Check the existence of Orderer ambassador certs secret in the vault 
 - name: Check if orderer ambassador secrets already created
   shell: |
-    vault kv get -format=yaml {{ vault.secret_path | default('secret') }}/crypto/{{ component_type }}Organizations/{{ component_name }}/ambassador/{{ orderer.name }}
+    vault kv get -field=certificate {{ vault.secret_path | default('secret') }}/crypto/{{ component_type }}Organizations/{{ component_name }}/ambassador/{{ orderer.name }}
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/create/crypto/orderer/tasks/orderer.yaml:175]: same comment as above for the get_ambassador_crypto role

**Review by sownak — CHANGES_REQUESTED** [review]
wherever `vault kv get -format=yaml` has been changed to `vault kv get -field=xyz`, the following get_crypto should be updated similar to how it is done in Besu. format=yaml returned the complete secret but field will only return one specific field

**Comment by sownak** [comment]
Also, get merge latest from vaultv2 so that the Besu files do not reappear

**Review by sownak — APPROVED** [review]

**Pull request merged**